ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT/ChainState.hs

```diff
@@ -353,14 +353,15 @@ append :: forall c. PBftCrypto c
        -> PBftChainState c -> PBftChainState c
 append k n signer@(PBftSigner _ gk) PBftChainState{..} =
     assertInvariant k n $
-    pruneEBBsLT $
     PBftChainState {
         preAnchor  = preAnchor'
       , postAnchor = postAnchor'
       , preWindow  = preWindow'
       , inWindow   = inWindow'
       , counts     = updateCounts counts
-      , ebbs       = ebbs   -- NB this needs to be pruned
+        -- NOTE: 'pruneEBBsLT' is inlined here to avoid a strange space leak
+        -- that also goes away with @-O0@, see #1356.
+      , ebbs       = EbbMap $ Map.filter (>= anchorSlot') (unEbbMap ebbs)
```

Comment:
That does seem odd. It seems like a real long shot, but the only other idea that comes to mind is: Does the space leak go away if you replace the record update expression in `pruneEBBsLT`'s definition with the explicit constructor?

Reply:
I have tried that, but it didn't help.

Reply:
@mrBliss FYI, if it's ever prioritized, I would enjoy debugging the GHC Core passes on this mystery.

Reply:
Noted! I had a look this morning, but the GHC Core (`-ddump-simpl`) version of `append` was ~1400 lines, so I quickly gave up :sweat_smile: